spec/requests/brands_controller_spec.rb

```diff
@@ -0,0 +1,258 @@
+      end
+    end
+
+    context 'when user logged in as manager' do
+      before(:each) do
+        sign_in manager
+      end
+
+      include_examples 'render index successfully'
+    end
+
+    context 'when user logged in as staff' do
+      before(:each) do
+        sign_in staff
+      end
+
+      include_examples 'render index successfully'
+    end
+
+    context 'when user logged out' do
+      it 'redirect to log-in page' do
+        get brands_path
+        expect(response).to redirect_to(new_account_session_path)
+      end
+    end
+  end
+
+  # brands#create
+  describe 'POST /brands/create' do
+    context 'when user logged in as manager' do
+      before(:each) do
+        sign_in manager
+      end
+
+      context 'with valid attributes' do
+        before(:each) do
+          post brands_create_path, params: { name: 'A new brand' }
+        end
+
+        it 'created successfully' do
+          expect(Brand.count).to eq(3)
+        end
+
+        it 'routed and rendered notice message successfully' do
+          expect(response).to redirect_to(brands_path)
+          follow_redirect!
+          expect(response.body).to include('New brand created successfully.')
+        end
+      end
+
+      context 'with invalid attributes' do
+        before(:each) do
+          post brands_create_path, params: { name: nil }
+        end
+
+        it 'created failed' do
+          expect(Brand.count).to eq(2)
+        end
+
+        it 'routed successfully' do
+          expect(response).to redirect_to(brands_path)
+        end
+      end
+
+      context 'with existed brand attributes' do
+        before(:each) do
+          post brands_create_path, params: attributes_for(:brand)
+        end
+
+        it 'created failed' do
+          expect(Brand.count).to eq(2)
+        end
+
+        it 'routed successfully' do
+          expect(response).to redirect_to(brands_path)
+        end
+      end
+    end
+
+    context 'when user logged in as staff' do
+      before(:each) do
+        sign_in staff
+        post brands_create_path, params: { name: 'A new brand' }
+      end
+
+      it 'created failed' do
+        expect(Brand.count).to eq(2)
+      end
+
+      it 'render unauthorized alert message' do
+        expect(response).to redirect_to(brands_path)
+        follow_redirect!
+        expect(response.body).to include('You are not authorized.')
+      end
+    end
+
+    context 'when user logged out' do
+      it 'redirect to log-in page' do
+        post brands_create_path, params: { name: 'A new brand' }
+        expect(response).to redirect_to(new_account_session_path)
+      end
+    end
+  end
+
+  # brands#update
+  describe 'PATCH /brands/update' do
+    context 'when user logged in as manager' do
+      before(:each) do
+        sign_in manager
+      end
+
+      context 'with valid attributes' do
+        before(:each) do
+          patch brands_update_path, params: { id: @brand2.id, new_name: 'A new brand name' }
```

Comment:
Cái này thường nằm trong subject
subject do
   patch brands_update_path, params: params
end

let(:params) {}

content 'when..' do
   let(:params) {}
   it 'abc' do
     subject
   end
end